packages/bloc/lib/src/cubit.dart

```diff
@@ -61,7 +61,7 @@ abstract class Cubit<State> extends Stream<State> {
 
   BlocObserver get _observer => Bloc.observer;
 
-  StreamController<State> _controller;
+  final StreamController<State> _controller = StreamController.broadcast();
```

Comment:
The state `StreamController` is crated conditionally in 3 different places, including `close`. This means that even though it is lazily created, it will be created at least once, when the cubit is closed. Why not create it right away? It will also be created only once, but the advantages are:

- It can be `final`.
- Easier code - no need for `_controller ??= ...` in 3 different places. Especially the one in `close` seems weird, but I guess it has to do with the fact that you want the `done` event to be emitted on `close`?

Reply:
I would prefer reverting this change because lazily creating the StreamController reduces the memory footprint of a bloc/cubit. While the controller will always be created, it will only be created when it is absolutely necessary instead of consuming unnecessary resources when it's not in use.

> but I guess it has to do with the fact that you want the done event to be emitted on close?
That's correct.

Reply:
I will do as you wish, but since you asked for my thoughts, here they are ;d

1. In Flutter, `BlocProvider` is lazy by default and created when the user first fetches it from the `BuildContext`. For the controller to be uninitialized when `Cubit.close` is called means that the dev did something like this: ```final b = context.read<SomeBloc>();``` and never, ever added any event, invoked a method changing the state or subscribed or whatever. This seems to be fishy code, likely a mistakte and I don't think the library needs to optimize for this case, code readability being the tradeoff.
2. In Flutter with an eager `BlocProvider`, the user needs to explicitly say they want it to be eager. If they do, eager should mean eager, so why would you make internal things lazy even in this case?
3. In Dart only, no Flutter - my arguments get weaker ;d but I think that if a user does ```final bloc = SomeBloc();``` and never uses it, it also means there is an issue in the code and there is no need to optimize in this case, similar to number 1 above.

Let me know and I will revert the `Cubit` changes.